generator/templates/schema.gotmpl

```diff
@@ -185,5 +185,40 @@ func ({{.ReceiverName}} *{{ pascalize .Name }}) UnmarshalBinary(b []byte) error
   *{{ .ReceiverName }} = res
   return nil
 }
-{{ end -}}
 
+// MarshalText interface implementation
+func ({{.ReceiverName}} *{{ pascalize .Name }}) MarshalText() ([]byte, error) {
+  if {{ .ReceiverName }} == nil {
+    return nil, nil
+  }
+  b, err := swag.WriteJSON({{ .ReceiverName }})
+  if err != nil {
+    return nil, err
+  }
+  return []byte(fmt.Sprintf("%v", b)), nil
+}
+
+// UnmarshalText interface implementation
+func ({{.ReceiverName}} *{{ pascalize .Name }}) UnmarshalText(text []byte) error {
+  var res {{ pascalize .Name }}
+  fmt.Printf("%s", string(text))
```

Comment:
this is not used, should use swag.ReadJSON

As a general rule, because I think you are trying to print to the log stream, fmt.Print  should not be used.

Reply:
fixed